### PR TITLE
Update HelpCommand to show correct description

### LINKED ISSUE #116: [PE-D][Tester E] Error in Help Panel Text (find)
In the section of the help panel brought up by the help command, the explanation of the find command is not up to date with the documentation:

"`find KEYWORD [MORE KEYWORDS]`" should be "`find n\KEYWORD [MORE_KEYWORDS]`, `find ic\KEYWORD`" instead

-------------
Labels: `severity.VeryLow` `type.FeatureFlaw`
original: tcjazwei/ped#5

### LINKED ISSUE #118: [PE-D][Tester E] Error in Help Panel Text (list)
In the section of the help panel brought up by the help command, the explanation of the find command is not up to date with the documentation:

"`list [WARD]` or `list [TAG]`" should be "`list [w\WARD] [t\TAG]...`" instead.

-------------
Labels: `severity.VeryLow` `type.FeatureFlaw`
original: tcjazwei/ped#6

### LINKED ISSUE #175: [PE-D][Tester C] Possible To Force Invalid Dates When Manually Editting File
Through editing the save file, it's possible to make DOB be Later than Admission date, as per screenshot:

To reproduce, simply change the save file to something like this:

This becomes an issue when trying to edit the person, since without editing their DOB, the edit command will always fail.

Though the UG mentions that editing the save file causes it to work in unexpected ways, it should probably prioritize resetting the save file so that the app can still function normally.

-------------
Labels: `severity.Medium` `type.FunctionalityBug`
original: drustanyjt/ped#16

### LINKED ISSUE #177: [PE-D][Tester C] Manually Editting Save Value of Ward Can Cause App To Crash On Start
The UG mentions that data files with invalid format will discard all data. Invalid Ward fields in the save file will however cause the program to crash on start instead.

To reproduce:
1. Invalidate the ward field of any person in the save file (for example "ward":"-3")
2. Try to run the app

The app will crash with this error message:

Rendering it completely unusable. Save filed for reference (line 8 has the offending change):

This does not seem to be a problem for any other field, leaving me to believe some validation check for this specific field is missing unintentionally.

-------------
Labels: `severity.High` `type.FunctionalityBug`
original: drustanyjt/ped#17

### PR DESCRIPTION
**Pull request by iamtr**
This PR will update the help command to display accurate information regarding the commands.
This also updates the relevant user guide portion.

_### **WE MIGHT NEED TO UPDATE THE OTHER PORTIONS OF THE UG TO REFLECT ON THE CHANGES HERE!!!**_

For example, 

Find: find n\NAME [MORE_NAMES] OR find ic\IC_NUMBER

but in UG:

image

Closes https://github.com/AY2324S2-CS2103T-F10-1/tp/issues/175
Closes https://github.com/AY2324S2-CS2103T-F10-1/tp/issues/177
Closes https://github.com/AY2324S2-CS2103T-F10-1/tp/issues/116
Closes https://github.com/AY2324S2-CS2103T-F10-1/tp/issues/118

### CODE REVIEW

**Comment by iamtr** [comment]
Closes #116
Closes #118

File: docs/UserGuide.md [hunk 1]
```diff
@@ -260,25 +260,33 @@ Output:
...
 Edit: edit INDEX [n\NAME] [ic\IC_NUMBER] [dob\DATE_OF_BIRTH] [ad\ADMISSION_DATE] [w\WARD] [r\REMARK] [t\TAG]...  
+   Edits the patient at the specified INDEX.
    Example: edit 1 ic\T0123456P t\
 
-Find: find KEYWORD [MORE_KEYWORDS]
-   Example: find John
-
-List: list
-   Lists all patients.
+Find: find n\NAME [MORE_NAMES] OR find ic\IC_NUMBER
```
Comment (ryanlimdx) [inline docs/UserGuide.md]: I think the [MORE NAMES] should be omitted:
image

**Comment by iamtr** [comment]
The following commits will amend the problem that the ward tag is invalid will cause the app to crash.

Closes #175 
Closes #177

File: docs/UserGuide.md (hunk 1 above)
Comment (iamtr) [reply docs/UserGuide.md]: Resolved

**Review by ryanlimdx — APPROVED** [review]
LGTM

**Pull request merged**